README.md
# d_button

Custom buttons which you can create and re-style as you wish. It can be used for one or more widgets.
Child of button can be filled with any widget as long as there is sufficient space.
But, for DButton.circle() just to use for layout circle.

You can modify all the buttons above to your liking and be as creative as possible

## Usage

1. Circle

<img src="https://github.com/indratrisnar/d_button/raw/master/pic/dbutton_circle.png">

```dart
DButtonCircle(
    mainColor: Colors.blue,
    onClick: () {},
    diameter: 40,
    child: Icon(
        Icons.favorite,
        color: Colors.white,
    ),
),
```
<hr>

2. Shadow

<img src="https://github.com/indratrisnar/d_button/raw/master/pic/dbutton_shadow.png">

```dart
DButtonShadow(
    mainColor: Colors.blue,
    splashColor: Colors.cyan,
    onClick: () => {},
    radius: 30,
    height: 50,
    child: Text(
        "D'Button Shadow 1",
        style: TextStyle(color: Colors.white),
    ),
),
```
<hr>

3. Flat

<img src="https://github.com/indratrisnar/d_button/raw/master/pic/dbutton_flat.png">

```dart
DButtonFlat(
    mainColor: Colors.blue,
    onClick: () {},
    child: Text(
        "D'Button Flat 1",
        style: TextStyle(color: Colors.white),
    ),
),
```
<hr>

4. Elevation

<img src="https://github.com/indratrisnar/d_button/raw/master/pic/dbutton_elevation.png">

```dart
DButtonElevation(
    mainColor: Colors.blue,
    onClick: () {},
    child: Text(
        "D'Button Elevation 1",
        style: TextStyle(color: Colors.white),
    ),
),
```
<hr>

5. Border

<img src="https://github.com/indratrisnar/d_button/raw/master/pic/dbutton_border.png">

```dart
DButtonBorder(
    borderColor: Colors.pink,
    mainColor: Colors.blue,
    radius: 0,
    onClick: () {},
    child: Text(
        "D'Button Border 1",
        style: TextStyle(color: Colors.white),
    ),
),
```
<hr>

6. Custom Child

Custom child can use for type button like shadow, flat, elevation & border

<img src="https://github.com/indratrisnar/d_button/raw/master/pic/dbutton_custom_child.png">

```dart
DButtonShadow(
    radius: 8,
    padding: EdgeInsets.only(right: 8),
    shadowColor: Colors.grey,
    onClick: () {},
    child: Row(
        children: [
        ClipRRect(
            borderRadius: BorderRadius.only(
                topLeft: Radius.circular(8),
                bottomLeft: Radius.circular(8),
            ),
            child: Image.network(
                'https://images.unsplash.com/photo-1590829197118-b0609523669d?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=400&q=80',
                width: 80,
                height: 80,
                fit: BoxFit.cover,
            ),
        ),
        SizedBox(width: 8),
        Expanded(
            child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
                Text(
                    'Title',
                    style: TextStyle(fontWeight: FontWeight.bold),
                ),
                SizedBox(height: 8),
                Text(
                    "Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
                    maxLines: 2,
                    overflow: TextOverflow.ellipsis,
                ),
            ],
            ),
        ),
        ],
    ),
),
```

Check at [Github](https://github.com/indratrisnar/d_button),

Support me for more feature & packages
[Donate](https://www.paypal.com/paypalme/indratrisnar)

Check my app : [Visit](https://indratrisnar.github.io/projects.html)

Check My Tutorial & Course : [Watch](https://www.youtube.com/channel/UC0d_xINEvCtlDCpWfBpnYpA)


CHANGELOG.md
## 0.0.10

Change call widget.

Before -> call via method
example: DButton.circle()

After -> call via class
example: DButtonCircle

## 0.0.3

- change format doc comment on button shadow

## 0.0.2

- Add documentation
- Add Button Border

## 0.0.1

DButton package :
- Button Circle
- Button Shadow
- Button Flat
- Button Elevation

Child can be filled with any widget as long as there is sufficient space.
But, for DButton.circle() just to use for layout circle.

You can modify all the buttons above to your liking and be as creative as possible

test/d_button_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:d_button/d_button.dart';

void main() {
  // test('adds one to input values', () {
  //   final calculator = Calculator();
  //   expect(calculator.addOne(2), 3);
  //   expect(calculator.addOne(-7), -6);
  //   expect(calculator.addOne(0), 1);
  // });
}


lib/d_button.dart
library d_button;

import 'package:flutter/material.dart';

/// Use static method to return custom button widget like [DButton.circle(child: Icon(Icons.favorite))]
class DButtonShadow extends StatelessWidget {
  /// [mainColor] body color of button. Default: Colors.white
  Color mainColor;

  /// [splashColor] color for splash effect. Default: Colors.grey
  Color splashColor;

  /// [shadowColor] color for shadow button. Default: Colors.black38
  Color shadowColor;

  /// [disableColor] color button if button disable / function onClick off. Default: Colors.grey
  Color disableColor;

  /// [onClick] action when button clicked
  Function()? onClick;

  /// [onDoubleClick] action when button double clicked
  Function()? onDoubleClick;

  /// [onLongClick] action when button long press
  Function()? onLongClick;

  /// [width] width of button
  double? width;

  /// [height] height of button
  double? height;

  /// [padding] padding of button
  EdgeInsetsGeometry? padding;

  /// [child] child of button. Widget can be anything as can as possible
  Widget child;

  /// [radius] radius for angle button. Default: 4.0
  double radius;

  /// [shadowOffset] Position of shadow button. Default: Offset.zero
  Offset shadowOffset;

  DButtonShadow({
    required this.child,
    this.disableColor = Colors.grey,
    this.height,
    this.mainColor = Colors.white,
    this.onClick,
    this.onDoubleClick,
    this.onLongClick,
    this.padding,
    this.radius = 4.0,
    this.shadowColor = Colors.black38,
    this.shadowOffset = Offset.zero,
    this.splashColor = Colors.grey,
    this.width,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      decoration: BoxDecoration(
        borderRadius: BorderRadius.circular(radius),
        boxShadow: [
          BoxShadow(
            blurRadius: 5,
            color: shadowColor,
            offset: shadowOffset,
          ),
        ],
      ),
      child: Material(
        borderRadius: BorderRadius.circular(radius),
        color: onClick == null && onDoubleClick == null && onLongClick == null
            ? disableColor
            : mainColor,
        child: InkWell(
          borderRadius: BorderRadius.circular(radius),
          splashColor: splashColor,
          onTap: onClick,
          onLongPress: onLongClick,
          onDoubleTap: onDoubleClick,
          child: Container(
            width: width,
            height: height,
            padding: width == null
                ? padding ??
                    EdgeInsets.symmetric(
                      horizontal: 16,
                      vertical: 8,
                    )
                : height == null
                    ? EdgeInsets.symmetric(
                        horizontal: 0,
                        vertical: 8,
                      )
                    : null,
            alignment: Alignment.center,
            child: child,
          ),
        ),
      ),
    );
  }
}

/// [DButtonFlat] return custom button which has flat
class DButtonFlat extends StatelessWidget {
  /// [mainColor] body color of button. Default: Colors.white
  Color mainColor;

  /// [splashColor] color for splash effect. Default: Colors.grey
  Color splashColor;

  /// [disableColor] color button if button disable / function onClick off. Default: Colors.grey
  Color disableColor;

  /// [onClick] action when button clicked
  Function()? onClick;

  /// [onDoubleClick] action when button double clicked
  Function()? onDoubleClick;

  /// [onLongClick] action when button long press
  Function()? onLongClick;

  /// [width] width of button
  double? width;

  /// [height] height of button
  double? height;

  /// [padding] padding of button
  EdgeInsetsGeometry? padding;

  /// [child] child of button. Widget can be anything as can as possible
  Widget child;

  /// [radius] radius for angle button. Default: 4.0
  double radius;

  DButtonFlat({
    required this.child,
    this.disableColor = Colors.grey,
    this.height,
    this.mainColor = Colors.white,
    this.onClick,
    this.onDoubleClick,
    this.onLongClick,
    this.padding,
    this.radius = 4.0,
    this.splashColor = Colors.grey,
    this.width,
  });

  @override
  Widget build(BuildContext context) {
    return Material(
      borderRadius: BorderRadius.circular(radius),
      color: onClick == null && onDoubleClick == null && onLongClick == null
          ? disableColor
          : mainColor,
      child: InkWell(
        borderRadius: BorderRadius.circular(radius),
        splashColor: splashColor,
        onTap: onClick,
        onLongPress: onLongClick,
        onDoubleTap: onDoubleClick,
        child: Container(
          width: width,
          height: height,
          padding: width == null
              ? padding ??
                  EdgeInsets.symmetric(
                    horizontal: 16,
                    vertical: 8,
                  )
              : height == null
                  ? EdgeInsets.symmetric(
                      horizontal: 0,
                      vertical: 8,
                    )
                  : null,
          alignment: Alignment.center,
          child: child,
        ),
      ),
    );
  }
}

/// return custom button which has elevation
class DButtonElevation extends StatelessWidget {
  /// [mainColor] body color of button. Default: Colors.white
  Color mainColor;

  /// [splashColor] color for splash effect. Default: Colors.grey
  Color splashColor;

  /// [disableColor] color button if button disable / function onClick off. Default: Colors.grey
  Color disableColor;

  /// [shadowColor] color for shadow button
  Color? shadowColor;

  /// [onClick] action when button clicked
  Function()? onClick;

  /// [onDoubleClick] action when button double clicked
  Function()? onDoubleClick;

  /// [onLongClick] action when button long press
  Function()? onLongClick;

  /// [width] width of button
  double? width;

  /// [height] height of button
  double? height;

  /// [elevation] elevation of button
  double? elevation;

  /// [padding] padding of button
  EdgeInsetsGeometry? padding;

  /// [child] child of button. Widget can be anything as can as possible
  Widget child;

  /// [radius] radius for angle button. Default: 4.0
  double radius;
  DButtonElevation({
    required this.child,
    this.disableColor = Colors.grey,
    this.elevation,
    this.height,
    this.mainColor = Colors.white,
    this.onClick,
    this.onDoubleClick,
    this.onLongClick,
    this.padding,
    this.radius = 4.0,
    this.shadowColor,
    this.splashColor = Colors.grey,
    this.width,
  });

  @override
  Widget build(BuildContext context) {
    return Material(
      shadowColor: shadowColor,
      elevation: elevation ?? 4,
      borderRadius: BorderRadius.circular(radius),
      color: onClick == null && onDoubleClick == null && onLongClick == null
          ? disableColor
          : mainColor,
      child: InkWell(
        borderRadius: BorderRadius.circular(radius),
        splashColor: splashColor,
        onTap: onClick,
        onLongPress: onLongClick,
        onDoubleTap: onDoubleClick,
        child: Container(
          width: width,
          height: height,
          padding: width == null
              ? padding ??
                  EdgeInsets.symmetric(
                    horizontal: 16,
                    vertical: 8,
                  )
              : height == null
                  ? EdgeInsets.symmetric(
                      horizontal: 0,
                      vertical: 8,
                    )
                  : null,
          alignment: Alignment.center,
          child: child,
        ),
      ),
    );
  }
}

/// return custom button circle
class DButtonCircle extends StatelessWidget {
  /// [mainColor] body color of button. Default: Colors.white
  Color mainColor;

  /// [splashColor] color for splash effect. Default: Colors.grey
  Color splashColor;

  /// [shadowColor] color for shadow button
  Color? shadowColor;

  /// [disableColor] color button if button disable / function onClick off. Default: Colors.grey
  Color disableColor;

  /// [onClick] action when button clicked
  Function()? onClick;

  /// [onDoubleClick] action when button double clicked
  Function()? onDoubleClick;

  /// [onLongClick] action when button long press
  Function()? onLongClick;

  /// [diameter] diameter of circle button
  double? diameter;

  /// [padding] padding of button
  EdgeInsetsGeometry? padding;

  /// [child] child of button. Widget can be anything as can as possible
  Widget child;

  /// [shadowOffset] Position of shadow button. Default: Offset.zero
  Offset shadowOffset;
  DButtonCircle({
    required this.child,
    this.diameter,
    this.disableColor = Colors.grey,
    this.mainColor = Colors.white,
    this.onClick,
    this.onDoubleClick,
    this.onLongClick,
    this.padding,
    this.shadowColor,
    this.shadowOffset = Offset.zero,
    this.splashColor = Colors.grey,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      decoration: BoxDecoration(
        shape: BoxShape.circle,
        boxShadow: shadowColor == null
            ? []
            : [
                BoxShadow(
                  blurRadius: 5,
                  color: shadowColor!,
                  offset: shadowOffset,
                ),
              ],
      ),
      child: Material(
        borderRadius: BorderRadius.circular(diameter ?? 1000),
        color: onClick == null && onDoubleClick == null && onLongClick == null
            ? disableColor
            : mainColor,
        child: InkWell(
          borderRadius: BorderRadius.circular(diameter ?? 1000),
          splashColor: splashColor,
          onTap: onClick,
          onLongPress: onLongClick,
          onDoubleTap: onDoubleClick,
          child: padding == null
              ? Container(
                  width: diameter,
                  height: diameter,
                  alignment: Alignment.center,
                  child: child,
                )
              : Padding(
                  padding: padding!,
                  child: child,
                ),
        ),
      ),
    );
  }
}

/// return custom button which has border
class DButtonBorder extends StatelessWidget {
  /// [mainColor] body color of button. Default: Colors.white
  Color mainColor;

  /// [borderColor] color of border
  Color borderColor;

  /// [borderWidth] sized of border width. Default: 2.0
  double borderWidth;

  /// [splashColor] color for splash effect. Default: Colors.grey
  Color splashColor;

  /// [disableColor] color button if button disable / function onClick off. Default: Colors.grey
  Color disableColor;

  /// [onClick] action when button clicked
  Function()? onClick;

  /// [onDoubleClick] action when button double clicked
  Function()? onDoubleClick;

  /// [onLongClick] action when button long press
  Function()? onLongClick;

  /// [width] width of button
  double? width;

  /// [height] height of button
  double? height;

  /// [padding] padding of button
  EdgeInsetsGeometry? padding;

  /// [child] child of button. Widget can be anything as can as possible
  Widget child;

  /// [radius] radius for angle button. Default: 4.0
  double radius;

  /// [shadowOffset] Position of shadow button. Default: Offset.zero
  Offset shadowOffset;
  DButtonBorder({
    required this.borderColor,
    this.borderWidth = 2.0,
    required this.child,
    this.disableColor = Colors.grey,
    this.height,
    this.mainColor = Colors.white,
    this.onClick,
    this.onDoubleClick,
    this.onLongClick,
    this.padding,
    this.radius = 4.0,
    this.shadowOffset = Offset.zero,
    this.splashColor = Colors.grey,
    this.width,
  });

  @override
  Widget build(BuildContext context) {
    return Material(
      borderRadius: BorderRadius.circular(radius),
      color: onClick == null && onDoubleClick == null && onLongClick == null
          ? disableColor
          : mainColor,
      child: InkWell(
        borderRadius: BorderRadius.circular(radius),
        splashColor: splashColor,
        onTap: onClick,
        onLongPress: onLongClick,
        onDoubleTap: onDoubleClick,
        child: Container(
          decoration: BoxDecoration(
            borderRadius: BorderRadius.circular(radius),
            border: Border.all(color: borderColor, width: borderWidth),
          ),
          width: width,
          height: height,
          padding: width == null
              ? height == null
                  ? EdgeInsets.symmetric(
                      horizontal: 16,
                      vertical: 8,
                    )
                  : padding
              : height == null
                  ? EdgeInsets.symmetric(
                      horizontal: 0,
                      vertical: 8,
                    )
                  : padding ?? null,
          alignment: Alignment.center,
          child: child,
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:d_button/d_button.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text('D\'Button')),
      backgroundColor: Colors.white,
      body: SingleChildScrollView(
        child: Padding(
          padding: EdgeInsets.all(16),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              ListTile(title: Text('Circle', style: TextStyle(fontSize: 18))),
              DButtonCircle(
                mainColor: Colors.blue,
                onClick: () {},
                diameter: 40,
                child: Icon(
                  Icons.favorite,
                  color: Colors.white,
                ),
              ),
              SizedBox(height: 16),
              FittedBox(
                child: DButtonCircle(
                  mainColor: Colors.white,
                  onClick: () {},
                  padding: EdgeInsets.all(8),
                  shadowColor: Colors.cyan,
                  child: Icon(
                    Icons.bookmark,
                    color: Colors.blue,
                    size: 15,
                  ),
                ),
              ),
              SizedBox(height: 16),
              DButtonCircle(
                mainColor: Colors.amber,
                onClick: () {},
                diameter: 30,
                child: Text('1'),
              ),
              SizedBox(height: 16),
              ListTile(title: Text('Shadow', style: TextStyle(fontSize: 18))),
              DButtonShadow(
                mainColor: Colors.blue,
                splashColor: Colors.cyan,
                onClick: () => {},
                radius: 30,
                height: 50,
                child: Text(
                  "D'Button Shadow 1",
                  style: TextStyle(color: Colors.white),
                ),
              ),
              SizedBox(height: 16),
              FittedBox(
                child: DButtonShadow(
                  mainColor: Colors.white,
                  shadowColor: Colors.blue,
                  padding: EdgeInsets.symmetric(horizontal: 16),
                  onClick: () => print('asasasasas'),
                  radius: 8,
                  child: Text("D'Button Shadow 2"),
                ),
              ),
              SizedBox(height: 16),
              DButtonShadow(
                mainColor: Colors.amber,
                shadowColor: Colors.blue,
                splashColor: Colors.yellow,
                onClick: () => print('asasasasas'),
                radius: 30,
                height: 30,
                width: 150,
                child: Text("D'Button Shadow 3"),
              ),
              SizedBox(height: 16),
              DButtonShadow(
                mainColor: Colors.amber,
                shadowColor: Colors.grey,
                splashColor: Colors.yellow,
                onClick: null,
                disableColor: Colors.grey[300]!,
                radius: 30,
                height: 30,
                width: 150,
                child: Text("D'Button Shadow 4"),
              ),
              SizedBox(height: 16),
              DButtonShadow(
                mainColor: Colors.white,
                shadowColor: Colors.blue,
                splashColor: Colors.yellow,
                onClick: () => {},
                radius: 30,
                height: 30,
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    Text("D'Button Shadow 5"),
                    SizedBox(width: 16),
                    Icon(Icons.arrow_forward),
                  ],
                ),
              ),
              SizedBox(height: 16),
              FittedBox(
                child: DButtonShadow(
                  mainColor: Colors.blue,
                  splashColor: Colors.yellow,
                  onClick: () => {},
                  padding: EdgeInsets.symmetric(horizontal: 30),
                  radius: 30,
                  height: 30,
                  child: Row(
                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                    children: [
                      Text("D'Button Shadow 6"),
                      SizedBox(width: 16),
                      Icon(Icons.arrow_forward),
                    ],
                  ),
                ),
              ),
              SizedBox(height: 16),
              DButtonShadow(
                mainColor: Colors.amber,
                splashColor: Colors.amberAccent,
                onClick: () => {},
                padding: EdgeInsets.symmetric(horizontal: 16),
                radius: 4,
                height: 40,
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    Text("D'Button Shadow 7"),
                    Icon(Icons.arrow_forward),
                  ],
                ),
              ),
              SizedBox(height: 16),
              DButtonShadow(
                mainColor: Colors.cyan,
                splashColor: Colors.blue,
                onClick: () => {},
                radius: 4,
                height: 40,
                width: 40,
                child: Icon(Icons.filter),
              ),
              SizedBox(height: 10),
              Divider(),
              ListTile(title: Text('Flat', style: TextStyle(fontSize: 18))),
              DButtonFlat(
                mainColor: Colors.blue,
                onClick: () {},
                child: Text(
                  "D'Button Flat 1",
                  style: TextStyle(color: Colors.white),
                ),
              ),
              SizedBox(height: 16),
              DButtonFlat(
                width: 150,
                radius: 150,
                mainColor: Colors.amber,
                onClick: () {},
                child: Text("D'Button Flat 2"),
              ),
              SizedBox(height: 16),
              FittedBox(
                child: DButtonFlat(
                  padding: EdgeInsets.symmetric(horizontal: 20, vertical: 8),
                  mainColor: Colors.white,
                  onClick: () {},
                  child: Text("D'Button Flat 3"),
                ),
              ),
              SizedBox(height: 16),
              FittedBox(
                child: DButtonFlat(
                  mainColor: Colors.blue,
                  onClick: null,
                  child: Text("D'Button Flat 4"),
                ),
              ),
              SizedBox(height: 16),
              DButtonFlat(
                mainColor: Colors.amber,
                onClick: () {},
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    Text("D'Button Flat 5"),
                    Icon(Icons.arrow_forward),
                  ],
                ),
              ),
              SizedBox(height: 16),
              FittedBox(
                child: DButtonFlat(
                  radius: 150,
                  height: 50,
                  mainColor: Colors.blue,
                  padding: EdgeInsets.symmetric(horizontal: 16),
                  onClick: () {},
                  child: Row(
                    children: [
                      Text("D'Button Flat 6"),
                      SizedBox(width: 16),
                      Icon(Icons.arrow_forward),
                    ],
                  ),
                ),
              ),
              SizedBox(height: 16),
              DButtonFlat(
                mainColor: Colors.cyan,
                onClick: () => {},
                radius: 4,
                height: 30,
                width: 30,
                child: Icon(Icons.search),
              ),
              SizedBox(height: 10),
              Divider(),
              ListTile(
                  title: Text('Elevation', style: TextStyle(fontSize: 18))),
              DButtonElevation(
                mainColor: Colors.blue,
                onClick: () {},
                child: Text(
                  "D'Button Elevation 1",
                  style: TextStyle(color: Colors.white),
                ),
              ),
              SizedBox(height: 16),
              DButtonElevation(
                elevation: 8,
                width: 150,
                radius: 150,
                mainColor: Colors.amber,
                onClick: () {},
                child: Text("D'Button Elevation 2"),
              ),
              SizedBox(height: 16),
              FittedBox(
                child: DButtonElevation(
                  elevation: 20,
                  padding: EdgeInsets.symmetric(horizontal: 20, vertical: 8),
                  mainColor: Colors.white,
                  onClick: () {},
                  child: Text("D'Button Elevation 3"),
                ),
              ),
              SizedBox(height: 16),
              FittedBox(
                child: DButtonElevation(
                  mainColor: Colors.blue,
                  onClick: null,
                  child: Text("D'Button Elevation 4"),
                ),
              ),
              SizedBox(height: 16),
              DButtonElevation(
                mainColor: Colors.amber,
                onClick: () {},
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    Text("D'Button Elevation 5"),
                    Icon(Icons.arrow_forward),
                  ],
                ),
              ),
              SizedBox(height: 16),
              FittedBox(
                child: DButtonElevation(
                  radius: 150,
                  height: 50,
                  elevation: 30,
                  mainColor: Colors.blue,
                  padding: EdgeInsets.symmetric(horizontal: 16),
                  onClick: () {},
                  child: Row(
                    children: [
                      Text("D'Button Elevation 6"),
                      SizedBox(width: 16),
                      Icon(Icons.arrow_forward),
                    ],
                  ),
                ),
              ),
              SizedBox(height: 16),
              DButtonElevation(
                mainColor: Colors.white,
                shadowColor: Colors.blue,
                onClick: () => {},
                elevation: 40,
                radius: 16,
                height: 50,
                width: 50,
                child: Icon(Icons.menu),
              ),
              SizedBox(height: 10),
              Divider(),
              ListTile(title: Text('Border', style: TextStyle(fontSize: 18))),
              DButtonBorder(
                borderColor: Colors.pink,
                mainColor: Colors.blue,
                radius: 0,
                onClick: () {},
                child: Text(
                  "D'Button Border 1",
                  style: TextStyle(color: Colors.white),
                ),
              ),
              SizedBox(height: 16),
              DButtonBorder(
                borderColor: Colors.blue,
                width: 150,
                radius: 150,
                mainColor: Colors.amber,
                onClick: () {},
                child: Text("D'Button Border 2"),
              ),
              SizedBox(height: 16),
              FittedBox(
                child: DButtonBorder(
                  borderColor: Colors.blue,
                  padding: EdgeInsets.symmetric(horizontal: 20, vertical: 8),
                  mainColor: Colors.white,
                  onClick: () {},
                  radius: 0,
                  child: Text("D'Button Border 3"),
                ),
              ),
              SizedBox(height: 16),
              FittedBox(
                child: DButtonBorder(
                  borderColor: Colors.amber,
                  mainColor: Colors.blue,
                  onClick: null,
                  child: Text("D'Button Border 4"),
                ),
              ),
              SizedBox(height: 16),
              DButtonBorder(
                borderColor: Colors.blue,
                mainColor: Colors.amber,
                onClick: () {},
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    Text("D'Button Border 5"),
                    Icon(Icons.arrow_forward),
                  ],
                ),
              ),
              SizedBox(height: 16),
              FittedBox(
                child: DButtonBorder(
                  borderColor: Colors.blue,
                  radius: 150,
                  height: 50,
                  mainColor: Colors.blue,
                  padding: EdgeInsets.symmetric(horizontal: 16),
                  onClick: () {},
                  child: Row(
                    children: [
                      Text("D'Button Border 6"),
                      SizedBox(width: 16),
                      Icon(Icons.arrow_forward),
                    ],
                  ),
                ),
              ),
              SizedBox(height: 16),
              DButtonBorder(
                borderColor: Colors.blue,
                mainColor: Colors.white,
                onClick: () => {},
                radius: 16,
                height: 50,
                width: 50,
                child: Icon(Icons.filter),
              ),
              SizedBox(height: 10),
              Divider(),
              ListTile(
                  title: Text('Custom Child', style: TextStyle(fontSize: 18))),
              DButtonShadow(
                radius: 8,
                padding: EdgeInsets.only(right: 8),
                shadowColor: Colors.grey,
                onClick: () {},
                child: Row(
                  children: [
                    ClipRRect(
                      borderRadius: BorderRadius.only(
                        topLeft: Radius.circular(8),
                        bottomLeft: Radius.circular(8),
                      ),
                      child: Image.network(
                        'https://images.unsplash.com/photo-1590829197118-b0609523669d?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=400&q=80',
                        width: 80,
                        height: 80,
                        fit: BoxFit.cover,
                      ),
                    ),
                    SizedBox(width: 8),
                    Expanded(
                      child: Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        mainAxisAlignment: MainAxisAlignment.center,
                        children: [
                          Text(
                            'Title',
                            style: TextStyle(fontWeight: FontWeight.bold),
                          ),
                          SizedBox(height: 8),
                          Text(
                            "Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
                            maxLines: 2,
                            overflow: TextOverflow.ellipsis,
                          ),
                        ],
                      ),
                    ),
                  ],
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
